### PR TITLE
Add open telemetry instrumentation to template files

### PR DESCRIPTION
**Pull request by HuwONS**
### What

Added open telemetry to the template generation functionality

### How to review

Check the templates generate as expected

### Who can review

Fran Moore, Chales Colbourn, Gordon Kerr

### Mentions
@HuwONS @gordonkerrONS @franmoore05 @hhsons @charlescolbourn

### CODE REVIEW

File: project_generation/content/templates/api/main.go.tmpl [hunk 1]
```diff
@@ -56,6 +59,22 @@ func run(ctx context.Context) error {
...
 
+	// Set up OpenTelemetry
+	otelConfig := dpotelgo.Config{
+		OtelServiceName:          cfg.OTServiceName,
+		OtelExporterOtlpEndpoint: cfg.OTExporterOTLPEndpoint,
+	}
+
+	otelShutdown, oErr := dpotelgo.SetupOTelSDK(ctx, otelConfig)
+	if oErr != nil {
+		log.Fatal(ctx, "error setting up OpenTelemetry - hint: ensure OTEL_EXPORTER_OTLP_ENDPOINT is set", oErr)
```
Comment (charlescolbourn) [inline project_generation/content/templates/api/main.go.tmpl:70]: Fatal & return - probably should not be the case, we don't want services to break if they don't have otel config

File: project_generation/content/templates/event/service/service.go.tmpl [hunk 2]
```diff
@@ -35,6 +37,8 @@ func Run(ctx context.Context, serviceList *ExternalServiceList, buildTime, gitCo
 
 	// Get HTTP Server with collectionID checkHeader middleware
 	r := mux.NewRouter()
+	otelHandler := otelhttp.NewHandler(router, "/")
+	router.Use(otelmux.Middleware(cfg.OTServiceName))
 	s := serviceList.GetHTTPServer(cfg.BindAddr, r)
 
 	// Get Kafka consumer
```
Comment (charlescolbourn) [inline project_generation/content/templates/event/service/service.go.tmpl:44]: check with @gordonkerrONS here. We want to have his kafka changes in, so maybe this needs to be blocked until that's merged & released.

File: project_generation/content/templates/event/service/service.go.tmpl [hunk 3]
```diff
@@ -35,6 +37,8 @@ func Run(ctx context.Context, serviceList *ExternalServiceList, buildTime, gitCo
 
 	// Get HTTP Server with collectionID checkHeader middleware
 	r := mux.NewRouter()
+	otelHandler := otelhttp.NewHandler(router, "/")
```
Comment (franmoore05) [inline project_generation/content/templates/event/service/service.go.tmpl:40]: When I generate a new event driven project, I get an error "undefined: router"

**Comment by franmoore05** [comment]
I think it would also be useful to instrument the controller generation (https://github.com/ONSdigital/dp-cli/tree/main/project_generation/content/templates/controller), as otel has been implemented in some of those already e.g. dp-frontend-homepage-controller

File: project_generation/content/templates/event/service/service.go.tmpl (hunk 3 above)
Comment (HuwONS) [reply project_generation/content/templates/event/service/service.go.tmpl]: I've fixed this and added in the instrumentation here now, but am having running the project including in develop

**Review by franmoore05 — APPROVED** [review]

**Pull request merged**